runtime/src/main/scala/akka/grpc/javadsl/GrpcMarshalling.scala

```diff
@@ -39,10 +39,7 @@ object GrpcMarshalling {
       u: ProtobufSerializer[T],
       mat: Materializer,
       reader: GrpcProtocolReader): CompletionStage[T] =
-    data.via(reader.dataFrameDecoder).map(u.deserialize).runWith(Sink.headOption[T], mat).thenCompose[T] { opt =>
-      if (opt.isPresent) CompletableFuture.completedFuture(opt.get)
-      else failure(new MissingParameterException())
-    }
+    data.via(reader.dataFrameDecoder).map(u.deserialize).runWith(SingleParameterSink.create[T](), mat)
```

Comment:
I just used `Sink.head` here instead and then just acted on the first element. We we only expect a single element, maybe that would be the simpler solution to `require` that the head just returns a single element.

Reply:
E.g. like this:

```suggestion
    data.via(reader.dataFrameDecoder).map(u.deserialize).runWith(Sink.seq[T], mat).thenCompose[T] { opt =>
      if (!opt.isEmpty) {
        require(opt.size == 1, s"Expected only a single parameter but got ${opt.size}")
        CompletableFuture.completedFuture(opt.get(0))
      }
      else failure(new MissingParameterException())
    }
```

Reply:
This seems to fix it for me.

Reply:
Hmm, it still reproduces for me even with this change (and `clean`ing), after a few tries.

Reply:
Oops, I incorrectly updated the reproducer - both this PR and the `Sink.seq` approach seem work (since both are pulling until completion instead of cancelling after the first (and only) element).

Reply:
Ah, great to have that mystery resolved at least :)